src/client/algod.js

```diff
@@ -21,17 +21,27 @@ function Algod(token, baseServer = "http://r2.algorand.network", port = 4180) {
      * @returns {Promise<*>}
      */
     this.status = async function () {
-        let res = await c.get("/v1/status");
-        return res.body;
+        try {
+            let res = await c.get("/v1/status");
+            return res.body;
+        } catch (e) {
+            e.message = e.response.error.message;
```

Comment:
Consider simply returning the entire error obj `throw e.response.error`. Right now the code breaks abstractions and makes some pieces of the error hidden (such as the error code)

Reply:
The proposed change adds an extra field to the `e` object with no deletion, whereas throwing just `e.response.error` would remove much (>90%, as measured by field count) of the error data. 

Here's an example error object that comes out from the currently-proposed change: https://gist.github.com/EvanJRichard/e3872fb953dbdb68607a18e2e0e0d385. It's the same error object that was being returned, with all data preserved (including error code), with an additional field `message` added.

Reply:
Now, when I understand the purpose of this PR, I think it should be closed. 
The information exist at the top level of the response returned (by examine your example and some that I ran personally). 

The `message` that the PR surfaces up already exist just under the name `text` and the statusCode is under the name `Status` (Although wasn't surfaced)